Jenkinsfile

```diff
@@ -1,158 +1,66 @@
+@Library('fxtest@1.3') _
 
-/** Tox environment */
-def environment = 'tests'
-
-/** Map of desired capabilities */
+/** Desired capabilities */
 def capabilities = [
   browserName: 'Firefox',
   version: '50.0',
   platform: 'Windows 10'
 ]
 
-/** Write capabilities to JSON file
- *
- * @param desiredCapabilities capabilities to include in the file
-*/
-def writeCapabilities(desiredCapabilities) {
-    def defaultCapabilities = [
-        build: env.BUILD_TAG,
-        public: 'public restricted'
-    ]
-    def capabilities = defaultCapabilities.clone()
-    capabilities.putAll(desiredCapabilities)
-    def json = JsonOutput.toJson([capabilities: capabilities])
-    writeFile file: 'capabilities.json', text: json
-}
-
-/** Run Tox
- *
- * @param environment test environment to run
-*/
-def runTox(environment) {
-  try {
-    wrap([$class: 'AnsiColorBuildWrapper']) {
-      withCredentials([[
-        $class: 'StringBinding',
-        credentialsId: 'SAUCELABS_API_KEY',
-        variable: 'SAUCELABS_API_KEY']]) {
-        withEnv(["PYTEST_ADDOPTS=${PYTEST_ADDOPTS} " +
-          "--driver=SauceLabs " +
-          "--variables=capabilities.json " +
-          "--color=yes"]) {
-          sh "tox -e ${environment}"
-        }
-      }
-    }
-  } catch(err) {
-    currentBuild.result = 'FAILURE'
-    throw err
-  } finally {
-    dir('results') {
-      stash environment
-    }
+pipeline {
+  agent any
+  options {
+    ansiColor()
+    timestamps()
+    timeout(time: 1, unit: 'HOURS')
   }
-}
-
-/** Send a notice to #fxtest-alerts on irc.mozilla.org with the build result
- *
- * @param result outcome of build
-*/
-def ircNotification(result) {
-  def nick = "fxtest${BUILD_NUMBER}"
-  def channel = '#fx-test-alerts'
-  result = result.toUpperCase()
-  def message = "Project ${JOB_NAME} build #${BUILD_NUMBER}: ${result}: ${BUILD_URL}"
-  node {
-    sh """
-        (
-        echo NICK ${nick}
-        echo USER ${nick} 8 * : ${nick}
-        sleep 5
-        echo "JOIN ${channel}"
-        echo "NOTICE ${channel} :${message}"
-        echo QUIT
-        ) | openssl s_client -connect irc.mozilla.org:6697
-    """
-  }
-}
-
-stage('Checkout') {
-  node {
-    timestamps {
-      deleteDir()
-      checkout scm
-      stash 'workspace'
-    }
+  environment {
+    PYTEST_ADDOPTS =
+      "-n=10 " +
+      "--color=yes " +
+      "--tb=short " +
+      "--driver=SauceLabs " +
+      "--variables=capabilities.json"
+    SAUCELABS_API_KEY = credentials('SAUCELABS_API_KEY')
   }
-}
-
-stage('Lint') {
-  node {
-    timestamps {
-      deleteDir()
-      unstash 'workspace'
-      sh 'tox -e flake8'
+  stages {
+    stage('Lint') {
+      steps {
+        sh "tox -e flake8"
+      }
     }
-  }
-}
-
-try {
-  stage('Test') {
-    node {
-      timeout(time: 1, unit: 'HOURS') {
-        timestamps {
-          deleteDir()
-          unstash 'workspace'
-          try {
-            writeCapabilities(capabilities)
-            withCredentials([[
```

Comment:
I spoke to @abayer and apparently this will be addressed by https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/110. We may need to hold off upgrading this pipeline to declarative for now...